### PR TITLE
Issue 667

### PR DESCRIPTION
**Pull request by tiago-loureiro**
With this PR we introduce the possibility of changing emails and resetting/changing the password for providers. In addition, provider tokens should also have a more limited lifetime which we should change in the config.

This is better viewed comparing it against a [different branch](https://github.com/wireapp/wire-server/compare/feature/cassandra_prefix_search...issue-667) - that PR should be merged in first.

### CODE REVIEW

File: services/brig/test/integration/API/Provider.hs [hunk 1]
```diff
@@ -8,7 +8,7 @@ module API.Provider (tests, Config) where
 
 import Bilge hiding (accept, timeout, head)
 import Bilge.Assert
-import Brig.Types
+import Brig.Types hiding (NewPasswordReset (..), CompletePasswordReset(..), EmailUpdate (..), PasswordReset (..), PasswordChange (..))
```
Comment (elverkilde) [inline services/brig/test/integration/API/Provider.hs]: Oh dear, that's a lot of hiding. Did you try qualifying one of the imports instead?

Comment (tiago-loureiro) [reply services/brig/test/integration/API/Provider.hs]: I thought the same... but importing qualified results in ~100 errors (either with `Brig.Types` or `Brig.Types.Provider`). Also importing only the needed types results in an even "uglier" import :/

**Comment by tiago-loureiro** [comment]
Fixed the PR after rebasing.

**Comment by elverkilde** [comment]
👍

**Pull request merged**